File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/permissions/permissionsBase.ts
/// <reference path="../base/iBaseEntity.ts" />
/// <reference path="../base/baseEntity.ts" />
/// <reference path="../../../helpers.ts" />
/// <reference path="./entityPermission.ts" />
/// <reference path="./objectPermission.ts" />

namespace mdBusinessLogic.dataAccess.entities.permissions {
    export class permissionsBase extends base.BaseEntity {
        public EntityPermissions: Array<entityPermission>;
        public ObjectPermissions: Array<objectPermission>;

        constructor(obj?: permissionsBase) {
            super();
            this.EntityPermissions = new Array<entityPermission>();
            this.ObjectPermissions = new Array<objectPermission>();
            if (obj !== undefined && obj != null) {
                this.EntityPermissions = obj.EntityPermissions;
                this.ObjectPermissions = obj.ObjectPermissions;
            }
        }
    }
}

File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/Permissions/PermissionsBase.cs
using MD.Tools.BaseDataAccess.Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities.Permissions
{
    public abstract class PermissionsBase : BaseEntity<long>
    {
        #region Attributes
        private List<EntityPermission> _entityPermissions;
        private List<ObjectPermission> _objectPermissions;
        #endregion

        #region Properties
        public List<EntityPermission> EntityPermissions
        {
            get
            {
                if(_entityPermissions == null)
                {
                    _entityPermissions = new List<EntityPermission>();
                }
                return _entityPermissions;
            }
            set => _entityPermissions = value;
        }
        public List<ObjectPermission> ObjectPermissions
        {
            get
            {
                if (_objectPermissions == null)
                {
                    _objectPermissions = new List<ObjectPermission>();
                }
                return _objectPermissions;
            }
            set => _objectPermissions = value;
        }
        #endregion
    }
}
